x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -101,6 +101,10 @@ private XPackSettings() {
     public static final Setting<Boolean> FIPS_MODE_ENABLED =
         Setting.boolSetting("xpack.security.fips_mode.enabled", false, Property.NodeScope);
 
+    /** Setting for enabling enrollment process; set-up by the es start-up script */
+    public static final Setting<Boolean> ENROLLMENT_ENABLED =
+        Setting.boolSetting("xpack.security.enrollment.enabled", false, Property.NodeScope);
```

Comment:
You could check for the value of this in `CreateEnrollmentToken#create`, eg `XPackSettings.ENROLLMENT_ENABLED.get(environment.settings())`, otherwise this is called anywhere.